src/Simulation/Simulators/ToffoliSimulator/ToffoliSimulator.cs

```diff
@@ -20,478 +17,29 @@ public enum ToffoliDumpFormat
-            // I if the axis is I, and illegal otherwise.
-            else if (power == 1)
-            {
-                switch (axis)
-                {
-                    case Pauli.PauliI:
-                        equivalentToX = false;
-                        safeToControl = false;
-                        break;
-                    case Pauli.PauliX:
-                        equivalentToX = true;
-                        safeToControl = false;
-                        break;
-                    case Pauli.PauliY:
-                    case Pauli.PauliZ:
-                        throw new InvalidOperationException($"The Toffoli simulator can only perform non-identity rotations around the X axis.");
-                }
-            }
-            // If power == 0, then we're a multiple of pi, possible an even multiple.
-            else
-            {
-                equivalentToX = false;
-                safeToControl = (numerator % 2 == 0);
-            }
-
-            return (equivalentToX, safeToControl);
-        }
-
-        /// <summary>
-        /// Mark a qubit as being borrowed, tracking its state before borrowing.
-        /// </summary>
-        /// <param name="q">The qubit being borrowed.</param>
-        internal void BorrowQubit(Qubit q)
-        {
-            Stack<bool> stack;
-            if (!borrowedQubitStates.TryGetValue(q.Id, out stack))
-            {
-                stack = new Stack<bool>();
-                borrowedQubitStates.Add(q.Id, stack);
-            }
-            stack.Push(State[q.Id]);
-        }
-
-        /// <summary>
-        /// Mark a group of qubits as being borrowed, tracking their states before borrowing.
-        /// </summary>
-        /// <param name="qs">The qubits being borrowed.</param>
-        internal void BorrowQubits(IEnumerable<Qubit> qs)
-        {
-            foreach (var q in qs)
-            {
-                BorrowQubit(q);
-            }
-        }
-
-        /// <summary>
-        /// Validate that a borrowed qubit is being returned in its original state.
-        /// </summary>
-        /// <param name="q">The qubit being returned.</param>
-        internal void ReturnQubit(Qubit q)
-        {
-            var stack = borrowedQubitStates[q.Id];
-            if (stack.Pop() != State[q.Id])
-            {
-                throw new ExecutionFailException("Borrowed qubit returned not in the original state");
-            }
+    public class ToffoliSimulator : QuantumProcessorDispatcher {
+        public const uint DEFAULT_QUBIT_COUNT = 64;
+        private ToffoliSimulatorProcessor processor;
 
-            if (stack.Count == 0)
-            {
-                borrowedQubitStates.Remove(q.Id);
-            }
-        }
+        // Whether an exception is thrown when releasing qubits that are in a
+        // non-zero state is configured as an argument to the constructor of
+        // ReversibleSimulator, defaulted to `true`.
```

Comment:
I'm failing to parse the grammatical structure of this comment. Maybe reword it?

Reply:
I will fix that one. It also still refers to the old name from samples. I'll scan the files for that.